src/consensus/aft/impl/execution_utilities.h

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+#pragma once
+
+#include "aft_state.h"
+#include "consensus/aft/raft_types.h"
+#include "consensus/pbft/pbft_requests.h"
+#include "enclave/rpc_map.h"
+
+namespace enclave
+{
+  class RPCSessions;
+  class RPCMap;
+}
+
+namespace aft
+{
+  class RequestMessage;
+
+  struct RequestCtx
+  {
+    std::shared_ptr<enclave::RpcContext> ctx;
+    std::shared_ptr<enclave::RpcHandler> frontend;
+  };
+
+  class AbstractExecutionUtilities
```

Comment:
Do we have multiple impls? If not, I'd prefer to avoid the Abstract* every time.